core/templates/components/forms/validators/is-regex-matched.filter.spec.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for Validator to check if input matches the provided
+ * regular expression.
+ */
+
+
+describe('isRegexMatched Filter', function() {
+  const filterName = 'isRegexMatched';
+
+  beforeEach(angular.mock.module('oppia'));
+
+  it('should exist', angular.mock.inject(function($filter) {
+    expect($filter(filterName)).not.toEqual(null);
+  }));
+
+  it('should pass if the string matches the given regular expression',
+    angular.mock.inject(function($filter) {
+      let filter = $filter(filterName);
+      let args = {
+        regexPattern: '(^https:\\/\\/.*)|(^(?!.*:\\/\\/)(.*))'
+      };
+      expect(filter('https://', args)).toBe(true);
+      expect(filter('https://any-string', args)).toBe(true);
+      expect(filter('https://www.oppia.com', args)).toBe(true);
+      expect(filter('www.oppia.com', args)).toBe(true);
+    }));
+
+  it('should fail if the string does not match the given regular expression',
```

Comment:
I think this being a generic regex filter, there could be some other smaller tests you could add. The HTTPS specific tests should probably go into sanitized-url-editor's spec as thats where this specific thing is used.

Reply:
I added a couple more tests with different regular expressions but I didn't really know how to move this test into sanitized-url-editor's spec.